components/list.tsx

```diff
@@ -0,0 +1,419 @@
+import {
+  RemixiconComponentType,
+  RiArrowDownSLine,
+  RiDraggable,
+  RiFile2Fill,
+} from "@remixicon/react";
+import {
+  ChangeEvent,
+  Children,
+  cloneElement,
+  createContext,
+  Fragment,
+  isValidElement,
+  ReactElement,
+  ReactNode,
+  useContext,
+  useState,
+} from "react";
+import Searchbox from "./searchbox";
+import { cn } from "./../lib/utils";
+import { AnimatePresence, motion } from "framer-motion";
+import LoadingSpinner from "./loading-spinner";
+import Checkbox from "./checkbox";
+
+export interface ListProps {
+  searchable?: boolean;
+  onSearchRequested?: (search: ChangeEvent<HTMLInputElement>) => void;
+  children: ReactNode;
+  className?: string;
+  isLoading?: boolean;
+  draggable?: boolean;
+  selectable?: boolean;
+  onDragged?: (props: {
+    id: number;
+    oldGroupId: string;
+    newGroupId: string;
+    oldPosition: number;
+    newPosition: number;
+  }) => void;
+}
+
+export interface ListGroupProps {
+  id: string;
+  title: string;
+  children: ReactNode;
+  draggable?: boolean;
+  containerClassName?: string;
+  contentClassName?: string;
+  selectable?: boolean;
+}
+
+export interface ListItemProps {
+  id?: number;
+  title?: string;
+  subtitle?: string;
+  imageUrl?: string;
```

Comment:
should be `iconUrl` to be consistent with other, which is using the wording/term `iconUrl`